nltk/tokenize/punkt.py

```diff
@@ -1266,9 +1266,10 @@ def train(self, train_text, verbose=False):
         """
         if not isinstance(train_text, str):
             return train_text
-        return PunktTrainer(
+        self._params = PunktTrainer(
             train_text, lang_vars=self._lang_vars, token_cls=self._Token
         ).get_params()
+        return self._params
```

Comment:
The `PunktTrainer.train` returns the parameters of type `PunktParameters` through the `.get_params()` function (which returns `self._params` if training is not `self._finalized`)

And in `finalize_training()`, the `_params` collocations and sent_starters will be cleared. 


By default after passing the `train_text` through `PunktTrainer.train`, the training will finalize. To disable that beahvior you'll have to do something like:

```
x = PunktTrainer()
x._finalized = False
x.train(train_text)

# Continue training on 2nd text.
x.train(train_text_n)

# Then when you want to finalize and get the parameters, after n-1 texts
x._finalized = True
x.train(train_text_n)
```

Reply:
Doing this will finalize the training at the first `train_text` pass

```
x = PunktTrainer(train_text)
```